src/render/texture_array_cache.rs

```diff
@@ -342,3 +344,24 @@ impl TextureArrayCache {
         }
     }
 }
+
+/// A system to remove any modified textures from the TextureArrayCache. Modified images will be
+/// added back to the pipeline, and so will be reloaded. This allows the TextureArrayCache to be
+/// responsive to hot-reloading, for example.
+pub fn remove_modified_textures(
+    modified_image_ids: Res<ModifiedImageIds>,
+    mut texture_cache: ResMut<TextureArrayCache>,
+) {
+    let texture_is_unmodified =
+        |texture: &TilemapTexture| !modified_image_ids.is_texture_modified(texture);
+
+    texture_cache
+        .textures
+        .retain(|texture, _| texture_is_unmodified(texture));
+    texture_cache
+        .meta_data
+        .retain(|texture, _| texture_is_unmodified(texture));
+    texture_cache.prepare_queue.retain(texture_is_unmodified);
+    texture_cache.queue_queue.retain(texture_is_unmodified);
+    texture_cache.bad_flag_queue.retain(texture_is_unmodified);
```

Comment:
only strange thing I noticed was we don't seem to actually be using `bad_flag_queue` anywhere. What is this for?

Reply:
fascinating. I don't think it has ever been used. It seems to have originated here: https://github.com/StarArawn/bevy_ecs_tilemap/commit/020d10ea3c54dfacd406c873940b8bfb3cf37654